nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -5622,14 +5625,29 @@ public ProcessorDiagnosticsEntity getProcessorDiagnostics(final String id) {
         nifiMetricsRegistry.setDataPoint(aggregateEvent.getBytesReceived(), "TOTAL_BYTES_RECEIVED",
                 instanceId, "RootProcessGroup", rootPGName, rootPGId, "");
 
+        //Add total task duration for root to the NiFi metrics registry
+        final StatusHistoryEntity rootGPStatusHistory = getProcessGroupStatusHistory(rootPGId);
+        final List<StatusSnapshotDTO> aggregatedStatusHistory = rootGPStatusHistory.getStatusHistory().getAggregateSnapshots();
+        final long taskDuration = aggregatedStatusHistory.isEmpty() ? 0L :
+                aggregatedStatusHistory.get(aggregatedStatusHistory.size() - 1).getStatusMetrics()
+                .get(ProcessGroupStatusDescriptor.TASK_MILLIS.getField());
+        nifiMetricsRegistry.setDataPoint(taskDuration, "TOTAL_TASK_DURATION",
+                instanceId, "RootProcessGroup", rootPGName, rootPGId, "");
```

Comment:
It would be helpful to define `RootProcessGroup` as a static constant and reuse where it is currently referenced.